Automattic/wp-calypso
# Plugins: Remove legacy unused components

#### Changes proposed in this Pull Request

With #59126 we no longer use some of the good old plugins components, so this PR gets rid of them.

It also resolves one of the #58453 items, because one of the removed components contained some `UNSAFE_` React lifecycle methods.

#### Testing instructions

* Verify the removed code is not in use and Calypso still builds fine.
* Verify all tests pass.
* Smoke test plugins.
